segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java

```diff
@@ -639,9 +639,6 @@ private int readInternal(SegmentHandle handle, byte[] buffer, long offset, int b
         //There is only one file per segment.
         FileStatus currentFile = findStatusForSegment(handle.getSegmentName(), true);
         try (FSDataInputStream stream = this.fileSystem.open(currentFile.getPath())) {
-            if (offset + length > stream.available()) {
-                throw new ArrayIndexOutOfBoundsException();
-            }
             stream.readFully(offset, buffer, bufferOffset, length);
```

Comment:
Maybe add a catch for EOFException here, log it as debug/trace and then simply return. The contract of `SyncStorage` indicates this method should return the number of bytes read, which is less than or equal to `length`